### ISSUE TITLE
Update HSS to latest version

### ISSUE THREAD

**Issue by tiwari3101** [infiniteorbits/hart-software-services#4]
When the latest version of the HSS is used on the EM/FM, the DDR training passes but the software payload does not boot. 

- [ ] Isolate the difference between the v0.99.33 and the latest version. 
- [ ] Get a software payload working with the latest version of the HSS.

**Comment by tiwari3101** [comment]
Tested the latest HSS (v2024.02.1) on the Icicle Kit with a payload compiled from the main (commit e78cb493bf0e1bf279e49fdca0b3dcf65fe4ffd4). The yaml file used is from the older version of hss. (v0.99.33). The payload boots successfully. 

A new payload was uploaded to sector 0 using CSP and on rebooting, the new payload boots successfully as well. 

There was an error noted in the HSS that does not yet affect the software: 


![Image](https://github.com/infiniteorbits/hart-software-services/assets/114018836/cad05086-6a70-4cbd-8f67-2565ae409610)

**Comment by tiwari3101** [comment]
Able to boot the software on the EM003. There are some 'errors' observed in the HSS output on UART0. 


![Image](https://github.com/user-attachments/assets/59cbd992-07e4-433e-8c34-2c2954d8eadb)

Meeting with Microchip on 15/07 confirmed that the logs were normal (both on the Icicle Kit and the EM003). This will be fixed in future releases.

**Comment by tiwari3101** [comment]
Parallel investigation regarding the slow DDR Training on all EMs with all versions of HSS. 
Performed tests as recommended by Microchip, logs attached below: 

[mpfs-mmuart-interrupt.txt](https://github.com/user-attachments/files/16263285/mpfs-mmuart-interrupt.txt)
[mpfs-hal-ddr-demo.txt](https://github.com/user-attachments/files/16263286/mpfs-hal-ddr-demo.txt) 

Comments by Microchip: 

- The DDR training issue persists over different versions of HSs. It has never been seen before. May point to a Hardware issue. 
- Yacine to open a ticket with the hardware team at Microchip to investigate further. 
- The issue with the software not booting may be coming from the same root cause as the slow DDR training,

**Comment by tiwari3101** [comment]
All the issues with the HSS as of date are documented here: https://docs.google.com/document/d/1o13Kvx4mcxVh3mEmBoWlDEgqCyaWU-ll3IJoYu3bFrg/edit?usp=sharing

**Comment by mannykoum** [comment]
Any updates on this?

**Comment by abelixrev** [comment]
I have updated the master branch with the HSS v2024.06 version from July 3rd, commit 88771e7729fb8cc05e6c0a12195ebb29028ae4d6.

I have created a branch from master and added the slot_selection feature: [HSS/v0.99.41-slot-selection](https://github.com/infiniteorbits/hart-software-services/tree/HSS/v0.99.41-slot-selection).

I have resolved the issue where sometimes the HSS loads the payload and boots the software, but the software doesn't start. To fix this, I modified services/mmc/mmc_api.c to ensure the eMMC always initializes with a clk_rate of 50MHz and in SDR mode (previously it was 200MHz and in HS200 mode).

Along with the above fix, I have also prevented the HSS from rebooting itself and performing the DDR training multiple times before booting the software by disabling the cold reboot functions.

In this version, some 'errors' observed in the HSS output on UART0 do not appear (it seems Microchip has resolved this in this version).

This version has been validated on EM002 and EM003 and works correctly.

Therefore, we can conclude that we have an updated HSS version that works with slot selection. The only remaining issue seems to be the time taken for DDR training.
